### PR TITLE
Add in OptiType (configbuilder, type) module(s)

### PR DESCRIPTION
**Pull request by apeltzer**
This is a (draft) PR to add in OptiType for the nf-core/hlatyping pipeline soon. 

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `<SOFTWARE>.version.txt` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

I will work on this in the next days :-)

### CODE REVIEW

**Comment by apeltzer** [comment]
Will get to it today I think, once quay.io is back from the dead 😆

**Comment by apeltzer** [comment]
This should have a `test` folder that I need push to the named emit channel `output` - I think its because `$prefix` isn't defined at that point ...?

**Comment by apeltzer** [comment]
So this runs fine locally, but not on pytest?

File: tests/software/optitype/test.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+- name: optitype test_optitype
+  command: nextflow run tests/software/optitype -entry test_optitype -c tests/config/nextflow.config
+  tags:
+    - type
```
Comment (KevinMenden) [inline tests/software/optitype/test.yml:4]:
```suggestion
    - optitype
```

**Review by KevinMenden — COMMENTED** [review]
The "tag" in the `pytest_software.yml` and in the `test.yml` has to match.
That should hopefully solve it

**Comment by KevinMenden** [comment]
Okay tests are running now but md5 sums are not reproducible 🤦  a classic for PDFs and reports.
A check for that is not yet implemented in `nf-core modules create-test-yml` 
So need to remove those md5 sums.

**Review by KevinMenden — APPROVED** [review]

**Pull request merged**